File: cos_upload.go
// COS上传工具 — Go语言版本
// 用法：echo "内容" | go run cos_upload.go fortune/2026/05/2026-05-27.txt
// 从stdin读取内容，上传到腾讯云COS指定路径

package main

import (
	"context"
	"fmt"
	"io"
	"net/http"
	"net/url"
	"os"
	"strings"

	"github.com/tencentyun/cos-go-sdk-v5"
)

const bucketURL = "https://openclaw-backup-tx-1251036673.cos.ap-beijing.myqcloud.com"

func getEnvOrExit(key string) string {
	v := os.Getenv(key)
	if v == "" {
		fmt.Fprintf(os.Stderr, "错误: 环境变量 %s 未设置\n", key)
		os.Exit(1)
	}
	return v
}

func main() {
	if len(os.Args) < 2 {
		fmt.Fprintln(os.Stderr, "用法: echo \"内容\" | cos_upload <cos-path>")
		os.Exit(1)
	}
	cosPath := os.Args[1]

	secretID := getEnvOrExit("TENCENT_COS_SECRET_ID")
	secretKey := getEnvOrExit("TENCENT_COS_SECRET_KEY")

	// 读取stdin内容
	content, err := io.ReadAll(os.Stdin)
	if err != nil {
		fmt.Fprintf(os.Stderr, "读取输入失败: %v\n", err)
		os.Exit(1)
	}

	u, _ := url.Parse(bucketURL)
	b := &cos.BaseURL{BucketURL: u}
	client := cos.NewClient(b, &http.Client{
		Transport: &cos.AuthorizationTransport{
			SecretID:  secretID,
			SecretKey: secretKey,
		},
	})

	_, err = client.Object.Put(context.Background(), cosPath, strings.NewReader(string(content)), nil)
	if err != nil {
		fmt.Fprintf(os.Stderr, "COS上传失败: %v\n", err)
		os.Exit(1)
	}

	fmt.Printf("  ☁️  已上传COS: openclaw-backup-tx-1251036673/%s\n", cosPath)
}

File: cos-upload.py
#!/usr/bin/env python3
"""COS上传工具 - 使用cos_python_sdk_v5上传文件"""
import sys, os
from qcloud_cos import CosConfig, CosS3Client

def upload(content, cos_path):
    secret_id = os.environ.get("TENCENT_COS_SECRET_ID")
    secret_key = os.environ.get("TENCENT_COS_SECRET_KEY")
    if not secret_id or not secret_key:
        print("COS密钥未设置", file=sys.stderr)
        return False

    config = CosConfig(
        Region="ap-beijing",
        SecretId=secret_id,
        SecretKey=secret_key,
        Token=None,
        Scheme="https"
    )
    client = CosS3Client(config)
    
    try:
        response = client.put_object(
            Bucket="openclaw-backup-tx-1251036673",
            Body=content.encode("utf-8"),
            Key=cos_path
        )
        print(f"  ☁️  已上传COS: openclaw-backup-tx-1251036673/{cos_path}")
        return True
    except Exception as e:
        print(f"  ⚠️  COS上传失败: {e}", file=sys.stderr)
        return False

if __name__ == "__main__":
    content = sys.stdin.read()
    cos_path = sys.argv[1] if len(sys.argv) > 1 else "test.txt"
    if not upload(content, cos_path):
        sys.exit(1)
